portallocator/portallocator.go

```diff
@@ -165,6 +165,14 @@ func (p *PortAllocator) ReleasePort(ip net.IP, proto string, port int) error {
 	if !ok {
 		return nil
 	}
+	for _, pr := range protomap[proto].portRanges {
```

Comment:
yeah sorry, you are right I thought there was an easier way.
If you find the port in the range, then can you break the loop? Also if last is already < than the release port also in this case do you have to update last?